crates/viewer/re_data_ui/src/transform_frames.rs

```diff
@@ -0,0 +1,219 @@
+use re_chunk_store::UnitChunkShared;
+use re_log_types::EntityPath;
+use re_types::{
+    ComponentDescriptor, TransformFrameIdHash, archetypes,
+    components::{self, TransformFrameId},
+};
+use re_ui::{HasDesignTokens as _, UiExt as _, UiLayout, icons};
+use re_viewer_context::{
+    Item, SystemCommand, SystemCommandSender as _, TransformDatabaseStoreCache, ViewerContext,
+};
+
+/// The max amount of ancestors we show before putting a '…'.
+const MAX_SHOWN_ANCESTORS: usize = 100;
+
+/// The max amount of ancestors we show in a tooltip.
+///
+/// We want this to be less because we don't use a scroll area for
+/// tooltips.
+const MAX_SHOWN_ANCESTORS_TOOLTIP: usize = 3;
+
+struct TransformFrameInfo {
+    frame_id: TransformFrameId,
+    source_entity: Option<EntityPath>,
+}
+
+pub struct TransformFramesUi {
+    frames: Vec<TransformFrameInfo>,
+
+    /// True if there are more transform frames than the ones shown here.
+    more: bool,
+}
+
+impl TransformFramesUi {
+    pub fn from_components(
+        ctx: &ViewerContext<'_>,
+        query: &re_chunk_store::LatestAtQuery,
+        transform_frame_descr: &ComponentDescriptor,
+        transform_frame_chunk: &UnitChunkShared,
+        entity_components: &[(ComponentDescriptor, UnitChunkShared)],
+    ) -> Option<Self> {
+        // All frame descriptor components, sorted by priority.
```

Comment:
it's not all of them anymore. it's a very select list and we should probably note down our learnings there on why we pick those :)